### ISSUE TITLE
Thrown exceptions don't return to calling method

### ISSUE THREAD

**Issue by IAmSilK** [rube200/OpenMod.Economy#2]
For actions sent to the Dispatcher, if an exception is thrown, the calling task will never continue as no result is set.

Here's my modified code for `DataStoreDatabase.UpdateBalanceAsync`:

```c#
            m_EconomyDispatcher.Enqueue(async () =>
            {
                Console.WriteLine("Updating balance");
                var data = await m_DataStore.LoadAsync<AccountsCollection>(TableName) ??
                           Activator.CreateInstance<AccountsCollection>();
                if (!data.Accounts.TryGetValue(uniqueId, out var balance)) balance = DefaultBalance;
                Console.WriteLine("Balance: " + balance);

                Console.WriteLine("Amount: " + amount);
                balance += amount;
                Console.WriteLine("New Balance: " + balance);
                if (balance < 0)
                    throw new NotEnoughBalanceException(StringLocalizer["economy:fail:not_enough_balance", balance]);

                Console.WriteLine("Saving");
                data.Accounts[uniqueId] = balance;
                await m_DataStore.SaveAsync(TableName, data);

                tcs.SetResult(balance);
            });
```

Example output:
```
Updating balance
Balance: 0.00
Amount: -100.00
New Balance: -100.00
```

As the new balance is below zero, we know the exception is being thrown.

I'm calling `UpdateBalanceAsync` from a UnturnedCommand's execute method which means `NotEnoughBalanceException` should be caught and handled properly, outputting to the user they can't afford this. This output never happens.

I tried encapsulating UpdateBalanceAsync within a try/catch statement and was able to further confirm the code never continues past calling `await m_EconomyProvider.UpdateBalanceAsync` where `m_EconomyProvider` is an instance of `DataStoreDatabase`.

**Comment by rube200** [comment]
I think this problem is fixed with this commit [15405ae](https://github.com/rube200/OpenMod.Economy/commit/15405ae33ce352b901b22f5f51eaf2845760560c). Could you test it?

**Comment by IAmSilK** [comment]
Calling task still won't resume/catch the exception

**Comment by IAmSilK** [comment]
Fixed by 63d63167
